src/NHibernate/Impl/SqlQueryImpl.cs

```diff
@@ -170,10 +172,12 @@ public override IList<T> List<T>()
 
 		public NativeSQLQuerySpecification GenerateQuerySpecification(IDictionary<string, TypedValue> parameters)
 		{
+			var allQuerySpaces=new HashSet<string>(querySpaces ?? new List<string>());
+			allQuerySpaces.UnionWith(GetSynchronizedQuerySpaces());
```

Comment:
`NativeSQLQuerySpecification` already does deduplication, so we dont' need to do it here.